client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
ok. name it `AutoSync`.

A blocked function is easy to control, and users could know that it is working easily. If the user wants to run it unblock, he could just go it. I refer http.Server.Serve for this design.